src/Cake.Common/Build/BuildSystem.cs

```diff
@@ -8,15 +11,19 @@ namespace Cake.Common.Build
     /// </summary>
     public sealed class BuildSystem
     {
-        private readonly IAppVeyorProvider _appVeyorProvider;
+        private readonly Lazy<bool> _isRunningOnBuildSystem;
```

Comment:
Would prefer to not have these evaluated lazily. No real performance overhead since the build system normally caches them. Also increases complexity.